File: elixir/bob/bob.exs
defmodule Bob do
  def hey(input) do
    input = input |> String.strip

    cond do
      input |> String.ends_with?("?") ->
        "Sure."
      input |> String.length == 0 ->
        "Fine. Be that way!"
      (input == input |> String.upcase) and (input != input |> String.downcase) ->
        "Whoa, chill out!"
      true ->
        "Whatever."
    end
  end
end

File: python/bob/bob.py
# -*- coding: utf-8 -*-


def hey(what):
    what = what.strip()

    if what.isupper():
        return 'Whoa, chill out!'
    elif what.endswith('?'):
        return 'Sure.'
    elif not what:
        return 'Fine. Be that way!'

    return 'Whatever.'